tests/Integration/Behaviour/Features/Context/Domain/AttributeGroupFeatureContext.php

```diff
@@ -28,57 +28,29 @@
 
 namespace Tests\Integration\Behaviour\Features\Context\Domain;
 
-use AttributeGroup as LegacyAttributeGroup;
 use Behat\Gherkin\Node\TableNode;
-use Language;
 use PHPUnit\Framework\Assert;
 use PrestaShop\PrestaShop\Core\Domain\Product\AttributeGroup\Query\GetAttributeGroupList;
 use PrestaShop\PrestaShop\Core\Domain\Product\AttributeGroup\Query\GetProductAttributeGroups;
 use PrestaShop\PrestaShop\Core\Domain\Product\AttributeGroup\QueryResult\AttributeGroup;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
 use RuntimeException;
 use Tests\Integration\Behaviour\Features\Context\Util\PrimitiveUtils;
 
 class AttributeGroupFeatureContext extends AbstractDomainFeatureContext
 {
-    /**
-     * @Given attribute group :reference named :name in :langIso language exists
-     *
-     * @param string $attributeGroupReference
-     * @param string $name
-     * @param string $langIso
-     */
-    public function assertNamedAttributeGroupExists(string $attributeGroupReference, string $name, string $langIso): void
-    {
-        $langId = (int) Language::getIdByIso($langIso);
-
-        if (!$langId) {
-            throw new RuntimeException(sprintf('Language by iso code "%s" was not found', $langIso));
-        }
-
-        $attributeGroups = LegacyAttributeGroup::getAttributesGroups($langId);
-        $foundGroupId = null;
-
-        foreach ($attributeGroups as $attributeGroup) {
-            if ($attributeGroup['name'] === $name) {
-                $foundGroupId = (int) $attributeGroup['id_attribute_group'];
-
-                break;
-            }
-        }
-
-        Assert::assertNotNull($foundGroupId, sprintf('Attribute group named "%s" was not found', $name));
-        $this->getSharedStorage()->set($attributeGroupReference, $foundGroupId);
-    }
-
     /**
      * @Given there is a list of following attribute groups:
      *
      * @param TableNode $tableNode
      */
-    public function assertAllAttributeGroups(TableNode $tableNode): void
+    public function assertAllAttributeGroupsForDefaultShop(TableNode $tableNode): void
```

Comment:
I know it represents quite some work, but since you added the ShopConstraint to this query maybe it would be worth adding some tests for specific shop and for all shops You can't really handle the creation with current steps so it would always be associated to the default shop

But since you have a step to handle the association between Attribute group and its shops, you create two groups, associate the first to shop2 the second to shop3 And check that depending on the shop the list is different, but when you get the list for all shops then they are all present